src/modules/twap/updaters/QuoteObserverUpdater.tsx

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from 'react'
+import { Field } from 'legacy/state/swap/actions'
+import { useTradeQuote } from 'modules/tradeQuote'
+import { useDerivedTradeState } from 'modules/trade/hooks/useDerivedTradeState'
+import { useUpdateCurrencyAmount } from 'modules/trade/hooks/useUpdateCurrencyAmount'
+
+export function QuoteObserverUpdater() {
+  const { state } = useDerivedTradeState()
```

Comment:
im a bit confused about how this works. 

- I can see `QuoteObserverUpdater` is a twap updater (in the TWAP module)
- depends on `useDerivedTradeState` to get the state --> the buy amount (in the TRADE module)
- on changes, it will update it, but it will do it using `useUpdateCurrencyAmount` (in the TRADE module)

My thinking is, the `trade` module exports the state, and a method to update it, and it expects other modules to watch changes on it and call the update method 🧐
Seems a bit strange dependencies.